### PR TITLE
feat(reflection): use message-id transcript state

### PR DESCRIPTION
**Pull request by devanshrj**
## Summary
- Move reflection transcript progress from `auto_cursor_line` to per-conversation backend message ids with lazy legacy-state migration.
- Track completed/reflected turn counters from transcript state so MemFS step-count reflection fires from unreflected turns instead of absolute reminder turn count.
- Serialize per-conversation transcript state writes with an in-process mutex plus file lock; corrupt/empty lock files are only reaped after a short grace period so in-flight lock writers do not wedge future acquisitions.
- Anchor cursors on backend message ids while starting payload ranges after the previous cursor line, which preserves live TUI user transcript rows that do not carry canonical backend ids.
- This PR intentionally excludes idle-time sweep/passive reflection settings.

## Validation
- [x] `bun test src/tests/cli/reflection-transcript.test.ts src/tests/cli/memoryReminder.test.ts src/tests/utils/fileLock.test.ts`
- [x] `bun test src/tests/cli/reflection-auto-launch-wiring.test.ts src/tests/websocket/listen-reflection-wiring.test.ts src/tests/telemetry/reflection-events-wiring.test.ts`
- [x] `bun test src/tests/reminders/listen-session-context.test.ts`
- [x] `bun run typecheck`
- [x] `bunx --bun @biomejs/biome@2.2.5 check --write src/cli/app/useSubmitHandler.ts src/cli/helpers/memoryReminder.ts src/cli/helpers/reflectionTranscript.ts src/reminders/engine.ts src/tests/cli/memoryReminder.test.ts src/tests/cli/reflection-transcript.test.ts src/websocket/listener/turn.ts src/tests/utils/fileLock.test.ts src/utils/fileLock.ts`
- [x] Cloud-backed live smoke: confirmed state files write `reflected_through_message_id`, step-count reflection launches from `turns_since_last_successful_reflection`, and payload ranges retain preceding user rows without canonical backend ids.

👾 Generated with [Letta Code](https://letta.com)

### CODE REVIEW

File: src/utils/fileLock.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    // Lock vanished between EEXIST and read - retry immediately.
+    return true;
+  }
+  let acquiredAt: unknown;
+  try {
+    acquiredAt = (JSON.parse(raw) as { acquiredAt?: unknown }).acquiredAt;
+  } catch {
+    acquiredAt = undefined;
+  }
+  if (typeof acquiredAt !== "number" || Date.now() - acquiredAt <= staleMs) {
```
Comment (amelia-letta) [inline src/utils/fileLock.ts]: A crash between `open` and `writeFile` (lines 50-51) leaves an empty lock file. `JSON.parse("")` throws → `acquiredAt = undefined` → `typeof undefined !== "number"` → returns `false` (not stale). The lock is permanently stuck until manual deletion — every subsequent acquisition times out after 10s.

Consider treating a non-parseable lock file as stale (it can only exist if the writer crashed mid-write):

```suggestion
  let acquiredAt: unknown;
  try {
    acquiredAt = (JSON.parse(raw) as { acquiredAt?: unknown }).acquiredAt;
  } catch {
    // Corrupt/empty lock file — writer crashed mid-write. Treat as stale.
    acquiredAt = -Infinity;
  }
  if (typeof acquiredAt !== "number" || Date.now() - acquiredAt <= staleMs) {
    return false;
  }
```

Comment (devanshrj) [reply src/utils/fileLock.ts]: Good catch. Addressed in 686aaf38: corrupt/empty lock files are now reaped after a short grace period so crashed acquisitions do not wedge the lock, but live acquisitions are not reaped during the open→write window. I also added a regression test for empty lock files and made failed acquisitions clean up their lock file best-effort.
